packages/client/src/runtime/core/engines/data-proxy/DataProxyEngine.ts

```diff
@@ -447,30 +460,23 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
   }
 
   private extractHostAndApiKey() {
-    const datasources = this.mergeOverriddenDatasources()
-    const mainDatasourceName = Object.keys(datasources)[0]
-    const mainDatasource = datasources[mainDatasourceName]
-    const dataProxyURL = this.resolveDatasourceURL(mainDatasourceName, mainDatasource)
+    const serviceURL = resolveDatasourceUrl({
+      inlineDatasources: this.inlineDatasources,
+      overrideDatasources: this.config.overrideDatasources,
+      clientVersion: this.clientVersion,
+      env: this.env,
+    })
 
     let url: URL
     try {
-      url = new URL(dataProxyURL)
+      url = new URL(serviceURL)
     } catch {
       throw new InvalidDatasourceError('Could not parse URL of the datasource', {
         clientVersion: this.clientVersion,
       })
     }
 
-    const { protocol, host, searchParams } = url
-
-    if (protocol !== 'prisma:') {
-      throw new InvalidDatasourceError(
-        'Datasource URL must use prisma:// protocol when --accelerate or --data-proxy are used',
-        {
-          clientVersion: this.clientVersion,
-        },
-      )
-    }
```

Comment:
This is not needed anymore because the decision to load the Data Proxy engine depends on the contents of the URL, so we are sure at this stage it will at least contain `prisma://...`.